### PR TITLE
Support struct initialization with named arguments

### PR DESCRIPTION
**Pull request by Shaikh-Ubaid**
fixes https://github.com/lcompilers/lpython/issues/1800.

### CODE REVIEW

File: src/lpython/semantics/python_ast_to_asr.cpp [hunk 1]
```diff
@@ -6612,14 +6687,9 @@ class BodyVisitor : public CommonVisitor<BodyVisitor> {
             tmp = nullptr;
             return ;
         }
-        // Keyword arguments handled in make_call_helper
-        #define parse_args() if( x.n_keywords == 0 ) { \
-            args.reserve(al, x.n_args); \
-            visit_expr_list(x.m_args, x.n_args, args); \
-        } \
```
Comment (Shaikh-Ubaid) [inline src/lpython/semantics/python_ast_to_asr.cpp:6619]: Here, I was overlooking the first line `if( x.n_keywords == 0 ) { \`. Hence, I refactored this into a function, which I am hoping is more readable.

**Comment by Shaikh-Ubaid** [comment]
Ready.

**Review by czgdp1807 — APPROVED** [review]

**Review by certik — APPROVED** [review]
Beautiful, thank you!

File: src/lpython/semantics/python_ast_to_asr.cpp [hunk 2]
```diff
@@ -893,6 +893,63 @@ class CommonVisitor : public AST::BaseVisitor<Struct> {
...
 
+    int64_t find_argument_position_from_name(ASR::StructType_t* orig_struct, std::string arg_name) {
+        int64_t arg_position = -1;
+        for( size_t i = 0; i < orig_struct->n_members; i++ ) {
+            std::string original_arg_name = std::string(orig_struct->m_members[i]);
+            if( original_arg_name == arg_name ) {
+                return i;
+            }
+        }
+        return arg_position;
```
Comment (Shaikh-Ubaid) [inline src/lpython/semantics/python_ast_to_asr.cpp:904]: I just noticed this. It seems `arg_position` is not needed here, we can just return `-1` at the end of the function. Sorry about it, I will update it in an upcoming PR.

Comment (Shaikh-Ubaid) [reply src/lpython/semantics/python_ast_to_asr.cpp:904]: fixed here https://github.com/lcompilers/lpython/pull/1816.

**Pull request merged**